src/main/java/org/quantumbadger/redreader/activities/AlbumListingActivity.java

```diff
@@ -136,12 +138,21 @@ public void onNotAnImage() {
 			@Override
 			public void onSuccess(final ImgurAPI.AlbumInfo info) {
 				Log.i("AlbumListingActivity", "Got album, " + info.images.size() + " image(s)");
-
-				AndroidApi.UI_THREAD_HANDLER.post(new Runnable() {
+				if(PrefsUtility.pref_behaviour_skipgallerylist(AlbumListingActivity.this, PreferenceManager.getDefaultSharedPreferences(AlbumListingActivity.this)) && !intent.hasExtra("viewList")) {
+					LinkHandler.onLinkClicked(
```

Comment:
Actually I think this could cause a bug -- if the activity is stopped while the album is loaded, the image will be shown to the user anyway.